#!/usr/bin/env python
PACKAGE = "mav_disturbance_observer"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("calibrate",    bool_t,    0, "remove constant offsets", False)

gen.add("q_position",    double_t,    0, "position noise", 0.1 ,  0.001, 10.0)
gen.add("q_velocity",    double_t,    0, "velocity noise", 0.1 ,  0.001, 10.0)
gen.add("q_attitude",    double_t,    0, "attitude noise", 0.1 ,  0.001, 10.0)
gen.add("q_angular_velocity",    double_t,    0, "angular velocity noise", 0.1 ,  0.001, 10.0)
gen.add("q_force",    double_t,    0, "external force noise", 0.1 ,  0.001, 10.0)
gen.add("q_torque",    double_t,    0, "external torque noise", 0.1 ,  0.001, 10.0)

gen.add("r_position",    double_t,    0, "position meas noise", 0.1 ,  0.001, 10.0)
gen.add("r_velocity",    double_t,    0, "velocity meas noise", 0.1 ,  0.001, 10.0)
gen.add("r_attitude",    double_t,    0, "attitude meas noise", 0.1 ,  0.001, 10.0)




exit(gen.generate(PACKAGE, "Config", "KFDisturbanceObserver"))
